### PR TITLE
IA-3097 delete OS disk and nic during VM deletion

### PR DESCRIPTION
**Pull request by Qi77Qi**
https://broadworkbench.atlassian.net/browse/IA-3097

See [this doc](https://docs.google.com/document/d/1yT3Io_FbOj_RybYl-rtvRAEcEuCgdmhCEMkQCta3XmY/edit) for a bit more context

Tested locally via swagger

### CODE REVIEW

**Review by ddietterich — COMMENTED** [review]
Huh. So the NIC and disk are implicitly created when the VM is created, but need to be explicitly deleted? Interesting.

Is it possible to make a test that would verify this is functioning as expected?

**Comment by Qi77Qi** [comment]
Yea. It's weird. Will add a test

**Comment by Qi77Qi** [comment]
@ddietterich I expanded the existing create Vm test a bit to include deletion....lmk what you think

**Review by ddietterich — APPROVED** [review]
LGTM - thank you!

File: service/src/test/java/bio/terra/workspace/service/resource/controlled/flight/CreateAndDeleteAzureControlledResourceFlightTest.java [hunk 1]
```diff
@@ -252,6 +256,47 @@ public void createAzureVmControlledResource() throws InterruptedException {
...
+    while (vmTemp == null) {
+      try {
+        vmTemp =
+            computeManager
+                .virtualMachines()
+                .getByResourceGroup(
+                    azureTestUtils.getAzureCloudContext().getAzureResourceGroupId(),
+                    creationParameters.getName());
+      } catch (com.azure.core.exception.HttpResponseException ex) {
+        if (ex.getResponse().getStatusCode() == 404) Thread.sleep(10000);
```
Comment (jdcanas) [inline service/src/test/java/bio/terra/workspace/service/resource/controlled/flight/CreateAndDeleteAzureControlledResourceFlightTest.java:272]: Seems a bit risky to not have an exit after a certain amount of tries here

**Review by jdcanas — APPROVED** [review]
small nit, otherwise looks good

**Pull request merged**